packages/webpack-cli/src/webpack-cli.ts

```diff
@@ -976,6 +976,16 @@ class WebpackCLI implements IWebpackCLI {
         description: "Stop watching when stdin stream has ended.",
         negatedDescription: "Do not stop watching when stdin stream has ended.",
       },
+      {
+        name: "fail-on-warnings",
+        configs: [
+          {
+            type: "boolean",
+          },
```

Comment:
Should we allow it to have a `number` type as well so that users can use their own exit code? Or we want to force the exit code as `1`.

`webpack --fail-on-warnings 2`

Reply:
Thanks for review, but I don't think we should. We have only 2 non-zero exit codes.
For exit code 2, we exit immediately by default. This option enforces immediate exit for `exitCode == 1`

Reply:
That sound reasonable to me 👍🏻. We can always support that in the future if required.